### ISSUE TITLE
e-Cert Implementation for Individual Student Processing

### ISSUE THREAD

**Issue by andrewsignori-aot** [bcgov/SIMS#2510]
Allow the processing of the student disbursements, individually processing its calculations and committing them to the database, defining them as "Ready to send" (new status to be added), which means, should be included in the next e-Cert generated. The news status would allow the e-Cert calculations and the file generation to happen at different times.

**Benefits**
- Allow each disbursement to be individually processed and committed. 
- Allow code simplifications that no longer need to deal with bulk processing.
- Allow the processing of multiple disbursements for the same students where account data as overawards deductions and maximums, must be updated (address https://github.com/bcgov/SIMS/pull/1886#discussion_r1170549440).
- In case one disbursement result has an issue it would not affect the other student disbursements.

**Acceptance Criteria**

- [x] Introduce a new `Ready to send` status for the `DisbursementScheduleStatus`. This status will be set when the calculations are done and the record should be part of the next e-Cert.
- [x] Add a new date column to capture the date when the `Ready to send` status was set. The suggested name `ready_to_send_date` in `sims.disbursement_schedules`.
- [x] Update the below reference codes to have the `Ready to Send` with the same effect as `Sent`, which means that SIMS will consider the money as sent either way (money was paid).
  - [x] MSFAA (minor): [getPreviouslySignedDisbursement](https://github.com/bcgov/SIMS/blob/674ad2fe07e0babe88014695fc2083247f635643/sources/packages/backend/apps/workers/src/services/disbursement-schedule/disbursement-schedule.service.ts#L167)
  - [x] IER12 (minor): [applicationEventCodeDuringCompleted](https://github.com/bcgov/SIMS/blob/bb71d47db285cb0c1ae76b53cb3116be33101db6/sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/utils-service/application-event-code-during-enrolment-and-completed-utils.service.ts#L59)
  - [x] Disbursement Schedule Creation (medium)
    - [sumDisbursedValuesPerValueCode](https://github.com/bcgov/SIMS/blob/185760482f5237dd55fc3bdf473b11ebbfca33a2/sources/packages/backend/libs/services/src/disbursement-schedule/disbursement-schedule-shared.service.ts#L422)
    - [totalDisbursedBCSLAmount](sources\packages\backend\libs\services\src\disbursement-schedule\disbursement-schedule-shared.service.ts#L691)
  - [x] e-Cert generation (minor): [prepareDisbursementsForECertGeneration](https://github.com/bcgov/SIMS/blob/4156444b840c6c09104dbd08d4a95754bc37c7bc/sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts#L99)
- [x] Adjust the current schedulers for `full-time-e-cert-integration` and `part-time-e-cert-integration` to execute the process with the proposed separation between calculations and the e-Cert generation.
  - [x] Create a process to execute the individual calculations
    - Select all disbursements not sent yet and not calculated, which means in 'Pending' status.
    - Ensure that only non-archived applications will be selected. Once the applications are archived the system should stop checking them, as per confirmed with @HRAGANBC.
    - Disbursement date in the past or in the near future.
    - Application status must be 'Completed'.
  - [x] For every select record, execute the calculation in parallel but ensure that records for the same student are processed sequentially by disbursement date ascendant.
      1. Create a DB transaction.
      2. Apply overawards deductions.
      3. Calculate effective value.
      4. Calculate the tuition remittance effective amount.
      5. Create BC Total grants award.
      6. Set the status to `Ready to Send`
      7. Commit the transaction.
  - [x] Create smaller services to handle the individual e-Cert steps. There is already a separation into methods and a further separation into services would make it easier to apply unit tests ([source code reference](https://github.com/bcgov/SIMS/blob/4156444b840c6c09104dbd08d4a95754bc37c7bc/sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts#L87-L101)).
  - [x] Allow selecting records to be processed with fewer conditions to generate a better individual log and easily capture the reason why a disbursement is not part of the e-Cert. For instance, allow students with canceled MSFAA or invalid SIN to be selected and execute an in-memory check to validate MSFAA and SIN, producing a log if necessary.
  - [x] Create scoped logs for every single student calculation processed and ensure that the output is sent to the POD log and job log.
  - [x] Create scoped logs for every single e-cert record generated and ensure that the output is sent to the POD log and job log.
